com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollStageAppStatusExecution.java

```diff
@@ -37,18 +43,32 @@ public AsyncExecutionState execute(ExecutionWrapper execution) {
         CloudControllerClient client = execution.getControllerClient();
 
         try {
+            stagingResolver.setClient(client);
```

Comment:
No. If you're going to change the state of a Spring component, then the component shouldn't be a component at all (there are some exceptions for prototype components though). Modifying a singleton that is used in multiple threads is **extremely** dangerous. What if two processes start using the same object at the same time. The second would override the client of the first and then the StagingResolver will work with the second client for both threads - thus giving one user control over the other user's space.